src/pages/Loginpage/index.tsx

```diff
@@ -0,0 +1,85 @@
+import React  from 'react';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import { Form } from 'react-bootstrap';
+import { Formik} from 'formik';
+import { initialValues } from 'pages/Loginpage/initialValues';
+import { validationSchema } from 'pages/Loginpage/validationSchema';
+import { Styled } from 'pages/Loginpage/Loginpage.styles';
+import CustomInputComponent from 'components/Input';
+import CustomButton from 'components/Button';
+import { login } from 'store/actions/auth';
+import { getAuthError, getIsAuthenticated, getAuthLoading } from 'store/selectors/auth';
+import { RootStore } from 'store';
+
+
+const Loginpage = ( props: any ) => {
+    const { loading, error, isAuthenticated } = props;
+
+    const submitHandler = (values: any) => {
+        props.loginHandler(values.username, values.password);
+    };
+
+    return(
+        <Styled.OuterContainer>
+            {isAuthenticated ? <Redirect to='/homepage' /> : null}
+            <Styled.InnerContainer>
+                <Styled.Header>User Login</Styled.Header>
+                <Formik
+                initialValues={initialValues}
+                validationSchema={ validationSchema }
+                onSubmit={(values) =>{submitHandler(values)}}
+                >
+                    {({handleSubmit, handleChange, values}) =>(
+                        <Styled.FormWrapper>
+                        <Form onSubmit={handleSubmit}>
+                            <CustomInputComponent
+                            label='Username'
+                            type='text'
+                            placeholder='Enter username'
+                            onChange={handleChange}
+                            value={values.username}
+                            name='username' />
```

Comment:
Why are you passing these things here. You are using useField inside CustomInputComponent
by your approach you can follow this.
https://formik.org/docs/api/useField

Reply:
My understanding of this is that we pass all the props which include Custom props and Formik props to the CustomInputComponent and the CustomInputComponent renders the custom input component according to those props. And we access different formik props using useField hook.
Please correct me if I am wrong.